src/chain/shared_state.rs

```diff
@@ -520,15 +520,21 @@ impl SectionProofBlock {
 }
 
 #[derive(Debug, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
-pub struct SectionProofChain {
-    genesis_key_info: SectionKeyInfo,
+pub struct SliceProofChain {
+    /// The version of the section key to use as root of trust.
+    version: u64,
+    /// The prefix of section key to use as root of trust.
+    prefix: Prefix<XorName>,
+    /// chain of trust to the section, if empty use root of trust.
     blocks: Vec<SectionProofBlock>,
 }
```

Comment:
I might be misunderstanding something, but doesn't this still contain the public keys in all but the first block? I though the idea was the public keys would be taken only from `their_key_infos` so they don't have to be included in the messages?

Reply:
Ah wait. I think I get it now. We might not have the latest keys, so they still need to be included in the message so we can update `their_key_infos` with it. It's fine then.

Reply:
Well the new key we don't know have to be included otherwise how do we learn of them.
But mostly what this mean is the root of trust is not included, we have to have it to verify the message. What this mean is for most messages for which blocks is empty, there is no public key included with the message.